### PR TITLE
Feat/functional components

### PR DESCRIPTION
**Pull request by beausmith**
All but the AppRoot component.

### CODE REVIEW

**Comment by beausmith** [comment]
not seeing test fail locally

**Comment by eventualbuddha** [comment]
FWIW this would be a `refactor: ` commit type since it doesn't change functionality, right?

**Comment by beausmith** [comment]
Correct @eventualbuddha. I still don't have this enum set memorized.

File: src/components/YesNoContest.tsx [hunk 1]
```diff
@@ -40,84 +47,73 @@ interface Props {
...
-export default class YesNoContest extends React.Component<Props, State> {
-  public context!: React.ContextType<typeof BallotContext>
-  public state = initialState
-  private scrollContainer = React.createRef<HTMLDivElement>()
+  const [isScrollable, setIsScrollable] = useState(false)
+  const [isScrollAtBottom, setIsScrollAtBottom] = useState(true)
+  const [isScrollAtTop, setIsScrollAtTop] = useState(true)
+  const [overvoteSelection, setOvervoteSelection] = useState<Optional<YesOrNo>>(
+    undefined
+  )
```
Comment (eventualbuddha) [inline src/components/YesNoContest.tsx:59]: You can simplify this by leaving out the initial value:

```suggestion
  const [overvoteSelection, setOvervoteSelection] = useState<YesOrNo>()
```

File: src/pages/ReviewPage.tsx [hunk 2]
```diff
@@ -356,160 +357,155 @@ class ReviewPage extends React.Component<RouteComponentProps, State> {
...
-                <Button
-                  className="scroll-up"
-                  big
-                  primary
-                  aria-hidden
-                  data-direction="up"
-                  disabled={isScrollAtTop}
-                  onPress={this.scrollContestChoices}
+            <ScrollableContentWrapper isScrollable={isScrollable}>
+              {(contests as Contests).map((contest, i) => (
```
Comment (eventualbuddha) [inline src/pages/ReviewPage.tsx:400]: Keeping this as it was is fine for now, but really this cast is unneeded:

```suggestion
              {contests.map((contest, i) => (
```

File: src/components/CandidateContest.tsx [hunk 3]
```diff
@@ -79,194 +86,162 @@ interface Props {
...
+const CandidateContest = ({ contest, parties, vote, updateVote }: Props) => {
+  const context = useContext(BallotContext)
+  const scrollContainer = useRef<HTMLDivElement>(null) // eslint-disable-line no-restricted-syntax
+
+  const [attemptedOvervoteCandidate, setAttemptedOvervoteCandidate] = useState<
+    OptionalCandidate
+  >(undefined)
+  const [candidatePendingRemoval, setCandidatePendingRemoval] = useState<
+    OptionalCandidate
+  >(undefined)
```
Comment (eventualbuddha) [inline src/components/CandidateContest.tsx:104]: As in `YesNoContest`, you can omit the default value:

```suggestion
  const [attemptedOvervoteCandidate, setAttemptedOvervoteCandidate] = useState<
    Candidate
  >()
  const [candidatePendingRemoval, setCandidatePendingRemoval] = useState<
    Candidate
  >()
```

**Review by eventualbuddha — APPROVED** [review]
Exciting!

**Comment by eventualbuddha** [comment]
> Correct @eventualbuddha. I still don't have this enum set memorized.

Yeah, I still look it up sometimes. For `feat` I ask myself "Does the end-user of this app/library have a new feature as a result of this change?". If yes, and that new feature isn't actually just a fix, then `feat` is appropriate.

**Review by benadida — APPROVED** [review]

File: src/components/YesNoContest.tsx (hunk 1 above)
Comment (beausmith) [reply src/components/YesNoContest.tsx]: thx. done.

File: src/pages/ReviewPage.tsx (hunk 2 above)
Comment (beausmith) [reply src/pages/ReviewPage.tsx]: thx. done.

File: src/components/CandidateContest.tsx (hunk 3 above)
Comment (beausmith) [reply src/components/CandidateContest.tsx]: thx. done.

**Pull request merged**